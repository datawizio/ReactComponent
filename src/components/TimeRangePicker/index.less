@import "~antd/es/date-picker/style/index";
@import "~antd/es/time-picker/style/index";
@import "../../styles/variables";

.ant-picker-dropdown .ant-picker-time-range-wrapper .ant-picker-ranges {
  width: auto;
}
