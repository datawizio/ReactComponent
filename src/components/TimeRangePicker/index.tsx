import React, { useContext } from "react";

import DatePicker from "../DatePicker";

import { TimeRangePickerProps } from "antd/es/time-picker";
import dayjs from "dayjs";
import ConfigContext from "../ConfigProvider/context";

export interface TimePickerProps
  extends Omit<TimeRangePickerProps, "value" | "onChange"> {
  fullWidth?: boolean;
  value?: { from: string; to: string };
  onChange?: (value: { from: string; to: string }) => void;
}

const TimeRangePicker = React.forwardRef<any, TimePickerProps>(
  ({ value, fullWidth, ...props }, ref) => {
    const { translate } = useContext(ConfigContext);
    const handleChange = (_, value) => {
      const [from, to] = value ? value : [null, null];
      props.onChange && props.onChange({ from, to });
    };
    return (
      //@ts-ignore
      <DatePicker.RangePicker
        {...props}
        onChange={handleChange}
        value={
          value && [
            //@ts-ignore
            value.from ? dayjs(value.from, props.format ?? "HH:mm") : null,
            //@ts-ignore
            value.to ? dayjs(value.to, props.format ?? "HH:mm") : null
          ]
        }
        //@ts-ignore
        dropdownClassName={props.popupClassName}
        picker="time"
        className={fullWidth ? "ant-picker-w100" : ""}
        mode={undefined}
        ref={ref}
        placeholder={[translate("START_TIME"), translate("END_TIME")]}
      />
    );
  }
);

export default TimeRangePicker;
