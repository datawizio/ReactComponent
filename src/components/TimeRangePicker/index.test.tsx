import "jsdom-global/register";
import React from "react";
import { shallow } from "enzyme";

import TimeRangePicker from "./index";

const mockProps = {
  fullWidth: true,
  value: { from: "2020-10-22T13:06:36.873Z", to: "2020-10-22T17:06:36.873Z" },
  onChange: jest.fn()
};

const setUp = (props?) => shallow(<TimeRangePicker {...props} />);

describe("TimeRangePicker component", () => {
  let component;

  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("TimeRangePicker rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("TimeRangePicker rendered correctly with empty props", () => {
    const wrapper = setUp();
    expect(wrapper).toMatchSnapshot();
  });

  it("TimeRangePicker trigger onChange with empty", () => {
    const rangePicker = component.find("RangePicker").first();
    rangePicker.props().onChange("");
    expect(component).toMatchSnapshot();
  });

  it("TimeRangePicker trigger onChange with mock", () => {
    const timeRange = ["2021-06-14T02:04:03.000Z", "2021-06-14T06:06:05.000Z"];
    const rangePicker = component.find("RangePicker").first();
    rangePicker.props().onChange(timeRange);
    expect(component).toMatchSnapshot();
  });
});
