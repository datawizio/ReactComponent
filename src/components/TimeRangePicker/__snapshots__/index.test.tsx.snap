// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeRangePicker component TimeRangePicker rendered correctly 1`] = `
<RangePicker
  className="ant-picker-w100"
  dropdownClassName="time-range-picker"
  onChange={[Function]}
  picker="time"
  placeholder={
    Array [
      "START_TIME",
      "END_TIME",
    ]
  }
  value={
    Array [
      "2021-12-08T20:00:00.000Z",
      "2021-12-08T20:00:00.000Z",
    ]
  }
/>
`;

exports[`TimeRangePicker component TimeRangePicker rendered correctly with empty props 1`] = `
<RangePicker
  className=""
  dropdownClassName="time-range-picker"
  onChange={[Function]}
  picker="time"
  placeholder={
    Array [
      "START_TIME",
      "END_TIME",
    ]
  }
/>
`;

exports[`TimeRangePicker component TimeRangePicker trigger onChange with empty 1`] = `
<RangePicker
  className="ant-picker-w100"
  dropdownClassName="time-range-picker"
  onChange={[Function]}
  picker="time"
  placeholder={
    Array [
      "START_TIME",
      "END_TIME",
    ]
  }
  value={
    Array [
      "2021-12-08T20:00:00.000Z",
      "2021-12-08T20:00:00.000Z",
    ]
  }
/>
`;

exports[`TimeRangePicker component TimeRangePicker trigger onChange with mock 1`] = `
<RangePicker
  className="ant-picker-w100"
  dropdownClassName="time-range-picker"
  onChange={[Function]}
  picker="time"
  placeholder={
    Array [
      "START_TIME",
      "END_TIME",
    ]
  }
  value={
    Array [
      "2021-12-08T20:00:00.000Z",
      "2021-12-08T20:00:00.000Z",
    ]
  }
/>
`;
